### DISCUSSION TITLE
What is the difference between computed and without computed?

### DISCUSSION THREAD

**Discussion by zestlee1106** [nuxt/nuxt#19232] [Questions] [answered]
```javascript
const a = 123
const test = a.value + 1
const computedTest = computed(() => a.value + 1)
```

what's defference?

**Comment by AloisSeckar** [comment]
I take it you meant `a = ref(123)` on your first line of code? Otherwise it doesn't make much sense. Assuming that:

- `test` will be "static" value that won't get updated in template if `a` is changed later
- `computedTest` will reactively update accordingly when `a` changes

Also without extra arguments a ref returned from `computed()` is read-only (see [Vue.js docs](https://vuejs.org/api/reactivity-core.html#computed))

**Reply by zestlee1106** [reply]
Thank you for the comments!

I wrote the wrong example.  
What's the difference in this case?
```javascript
const a = ref(123)
const test = a.value + 1
const computedTest = computed(() => a.value + 1)
```

**Comment by ndragun92** [comment]
What is the difference between methods and computed in Vue?
A computed property is **cached**, which implies that, for accessing data, the function runs just once if the value remains unchanged. On the other hand, methods need to run data property checks every time because they cannot tell otherwise if the values are changed inside the method property.

But in your example, `a` and `test` are statically declared variables

If you for example had
`const a = ref(123)` Then you would not need to have it wrapper in computed because variable is reactive

**Reply by zestlee1106** [reply]
Thank you for the comments!

I wrote the wrong example.
What's the difference in this case?
```javascript
const a = ref(123)
const test = a.value + 1
const computedTest = computed(() => a.value + 1)
```

**Reply by ndragun92** [reply]
a is reactive test is not reactive variable

**Comment by TheAlexLichter** [comment, marked as answer]
```ts
const a = ref(1)
const test = a.value + 1
const computedTest = computed(() => a.value + 1)
a.value = 3

console.log(test) // 2 because `test` is not reactive
console.log(computedTest.value) // 4 because the computed value will be updated
```
